asterius/src/Asterius/GHCi/Internals.hs

```diff
@@ -287,7 +287,9 @@ asteriusWriteIServ hsc_env i a
                 linkOutput = "",
                 linkObjs = [],
                 linkLibs = [],
-                linkModule = m <> M.foldr' (<>) (ghciLibs s) (ghciObjs s),
+                linkModule =
```

Comment:
IMO this will impact TH performance a lot; the `linkModule` field we pass to the linker logic contains all the bundled archives/objects without the symbol graph info. And the linker will reconstruct the symbol graph on the fly, which is wasteful. I can see two different fixes:

* Make `linkModule` an `AsteriusCachedModule` too, and don't throw away the loaded symbol graph by calling `fromCachedModule` here
* Change the `ghciLibs` and `ghciObjs` representation; we no longer store the decoded modules in-memory, instead we'll simply store their file paths, and put the file paths in the `linkObjs`/`linkLibs` field here when calling `linkExeInMemory`.

Reply:
I see. I'll have a go at the first approach and see how it goes.